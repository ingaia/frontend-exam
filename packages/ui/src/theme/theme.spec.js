import React from 'react';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import { Theme, useTheme, withTheme } from './index';
import setTheme from './set-theme';

expect.addSnapshotSerializer(serializer);

test('renders theme with correct styles', () => {
  const theme = renderer.create(
    <Theme setGlobalStyles theme="dark">
      Testing
    </Theme>
  ).toJSON();

  expect(theme).toMatchSnapshot();
});

test('components can access theme prop', () => {
  const mock = setTheme();
  let result;

  const Comp = () => {
    const t = useTheme();

    result = t;

    return(<div>Test</div>);
  };

  const tree = renderer.create(<Theme><Comp /></Theme>);

  expect(mock).toEqual(result);
});
