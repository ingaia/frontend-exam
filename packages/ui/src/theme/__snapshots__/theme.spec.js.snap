// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders theme with correct styles 1`] = `"Testing"`;
